src/core/lwt.mli

```diff
@@ -21,502 +22,1785 @@
+      value, its callbacks are called.
+    - Separate {b resolvers} of type ['a ]{!Lwt.u} are used to write values into
+      promises, through {!Lwt.wakeup_later}.
+    - Promises and resolvers are created in pairs using {!Lwt.task}. Lwt I/O
+      functions call {!Lwt.task} internally, but return only the promise.
+    - {!Lwt_main.run} is used to wait on one “top-level” promise. When that
+      promise gets a value, the program terminates.
+
+
+
+    {3 Tutorial}
+
+    Let's read  from STDIN. The first version is written using ordinary values
+    from the OCaml standard library. This makes the program block until the user
+    enters a line:
+
+{[
+let () =
+  let line : string =
+    Pervasives.read_line () in
+  print_endline "Now unblocked!";
+  ignore line
+
+(* ocamlfind opt -linkpkg code.ml && ./a.out *)
+]}
+
+    If we use a promise instead, execution continues immediately:
+
+{[
+let () =
+  let line_promise : string Lwt.t =
+    Lwt_io.(read_line stdin) in
+  print_endline "Execution just continues...";
+  ignore line_promise
+
+(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
+]}
+
+    Indeed, this program is a little {e too} asynchronous – it exits right away!
+    Let's force it to wait for [line_promise] at the end by calling
+    {!Lwt_main.run}:
+
+{[
+let () =
+  let line_promise : string Lwt.t =
+    Lwt_io.(read_line stdin) in
+  print_endline "Execution just continues...";
+
+  let line : string =
+    Lwt_main.run line_promise in
+  ignore line
+
+(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
+]}
+
+    {!Lwt_main.run} should only be called once, on one promise, at the top level
+    of your program. Most of the time, waiting for promises is done using
+    [let%lwt]. That is the recommended syntactic sugar for {!Lwt.bind}, and is
+    pronounced “bind”:
+
+{[
+let () =
+  let p : unit Lwt.t =
+    let%lwt line_1 = Lwt_io.(read_line stdin) in
+    let%lwt line_2 = Lwt_io.(read_line stdin) in
+    Lwt_io.printf "%s and %s\n" line_1 line_2
+  in
+
+  Lwt_main.run p
+
+(* ocamlfind opt -linkpkg -package lwt.ppx,lwt.unix code.ml && ./a.out *)
+]}
+
+    The way that works is everything in scope after the “[in]” in
+    “[lwt%lwt x =] ... [in] ...” goes into a callback, and “[x]” is that
+    callback's argument. So, we could have been very explicit, and written the
+    code like this:
+
+{[
+let () =
+  let p : unit Lwt.t =
+    let line_1_promise : string Lwt.t = Lwt_io.(read_line stdin) in
+    Lwt.bind line_1_promise (fun (line_1 : string) ->
+
+      let line_2_promise : string Lwt.t = Lwt_io.(read_line stdin) in
+      Lwt.bind line_2_promise (fun (line_2 : string) ->
+
+        Lwt_io.printf "%s and %s\n" line_1 line_2))
+  in
+
+  Lwt_main.run p
+
+(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
+]}
+
+    But, as you can see, this is verbose, and the indentation gets a bit crazy.
+    So, we will always use [lwt%lwt].
```

Comment:
Maybe show the bind operator (`>>=`) version?

```
+{[
let () =
  let p : unit Lwt.t =
    Lwt_io.(read_line stdin) >>= fun line_1 ->
    Lwt_io.(read_line stdin) >>= fun line_2 ->
    Lwt_io.printf "%s and %s\n" line_1 line_2
  in

  Lwt_main.run p

(* ocamlfind opt -linkpkg -package lwt.unix code.ml && ./a.out *)
]}
```

Reply:
Instead of this?

Just background:

I decided to offer the PPX for everything by default. To save space, and avoid needlessly offering too many options, I mention a `>>=` version of some code in the docs of `Lwt.bind` only.